connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -320,6 +320,10 @@ public SchemaBuilder field(String fieldName, Schema fieldSchema) {
         if (type != Type.STRUCT)
             throw new SchemaBuilderException("Cannot create fields on type " + type);
         int fieldIndex = fields.size();
+        for (Field elements : fields) {
```

Comment:
Iterating over the entire list on every added field looks a bit suboptimal. Since we are using this in the Converters and we convert every batch of messages, this may have actual impact.

Did you check what would it take to change the data structure of fields from List to Map (using field name as a key) and then use containsKey to check if the field already exists? 

It should be more efficient, but if this has too much impact on the code elsewhere, maybe it isn't worth it.